pkg/download/protocol.go

```diff
@@ -42,8 +42,16 @@ type Opts struct {
 	Stasher      stash.Stasher
 	Lister       module.UpstreamLister
 	DownloadFile *mode.DownloadFile
+	NetworkMode  string
```

Comment:
maybe make this its own type? obv not required, but could be helpful for readability

Reply:
I think it's easier as a string since it's only used internally and no need to introduce cyclical imports since this is being passed from the config into the download protocol